### DISCUSSION TITLE
Sepheo documentation

### DISCUSSION THREAD

**Discussion by shles** [sepheocoop/documentation#9] [Ideas]
@steve52 and I discussed how we can improve the documents for the Sepheo coop.
We outlined the  main goal for the documents:
**Any new member should understand the organization they are considering joining. It’s possible by exploring and easily navigating documentation, bylaws, meeting notes, and connected materials**
We want to find a way to organize these materials that is:
1. Easy to maintain, edit, and expand by the members
2. Easy to host as a part of the website
3. Everyone can see the history of edits

Here is the plan for achieving the first iteration of this goal.
1. Create a separate public repo with the part of the documentation that can be public. We can collectively discuss what parts should be public.
2. Move everything from Google Docs to this public-doc repo
    1.  Move Matchmaker to the documentation If it is not done, move it to the GitHub discussion and continue the discussion there 
3. Update the “Overview” doc. It should be a core document describing the main concepts in the coop.
    1. Add links to the separate documents that expand the explanation such as the Matchmaker doc
    2. Add content navigation to the Overview doc
4. Add goals as a separate document. This doc will contain logs of every quarter's goals with corresponding results.  This doc should be mentioned in the overview doc.

The next step could be setting up a wiki page (such as https://wiki.coops.tech/wiki/CoTech_Constitution) so that everything mentioned above is more organized and familiar to the public.
We can also create a more custom structure like this one https://www.loomio.coop/cooperative_principles.html
This is also up for group discussion. 

I can take the first step by organizing a new repo. I might need more permissions to edit current repos, or I can just copy everything in a new one.

Let's put the question to the discussion agenda:
- What should be made public
- What ultimate presentation for this info to use (wiki or custom)

**Comment by shles** [comment]
here is a good example of finished bylaws 

https://github.com/colab-coop/bylaws/tree/master

**Comment by King-Mob** [comment]
Goal 2 is complete. There's nothing on google docs that isn't on the github org

**Comment by King-Mob** [comment]
Goal 1: I recommend using the website repo, sepheo.co, to host public documentation: https://github.com/sepheocoop/website/discussions

That's where I put my discussion about what success might mean for example

**Reply by King-Mob** [reply]
The only thing I wouldn't make public is the notes on setting up cloudron. Everything else looks fine to be public.

**Reply by King-Mob** [reply]
If you put the docs in the website repo, then they can be easily rendered on the site, and it simplifies the pipeline from "Making Decision" to "Outcome publicly available"

**Reply by King-Mob** [reply]
although I've kind of muddied this by using the discussions part of the repository and not actually hosting a markdown document there. ah well.

**Reply by shles** [reply]
To clarify, should the docs be hosted in the repo itself or in the discussions? 

From the @wu-lee comment, I agree with the idea of tracking changes through the commits and PRs, which means using the repo itself. 

Then, what should discussions be used in regard to the docs that we are making public?

What needs to be done to render the docs on the website?

**Comment by wu-lee** [comment]
> * Easy to maintain, edit, and expand by the members
> 
> * Easy to host as a part of the website
> 
> * Everyone can see the history of edits

Totally agree on these objectives. I think these were pretty much what we had in mind with the stuff currently in the documentation repository, although we were erring on the side of caution by not making it public yet - I think it's useful to have somewhere to formulate ideas before they go live, although perhaps we should do that on a branch as we're using version control.

I think it's a particularly good idea to use version control for these documents, as I think in Code-operative we didn't, and this was a mistake. We used Google Docs, and its document tracking isn't really up to the job, doesn't work cross-document, and authorship and ownership can be unclear. Word or Libre Office don't do much better IMO.

Being able to discuss the history of edits (as in a pull request) is also very useful. As is to be able to say "this edit is all over the place, please could you separate it into cosmetic changes, and actual semantic changes."

One other point: as well as to use version control, for more formal documentation we decided to use AsciiDoc. (You'll see some files with an `.adoc` extension [in the repository][2].) This is a bit like Markdown, but is designed for writing books and documentation. As such it allows a lot more control over things like pagination, layout, numbering, references and the like. I'm suggesting this is at contention here, but nevertheless I want to draw attention to the fact that it's used, and encourage us to continue to use it for bylaws etc.

[2]: https://github.com/sepheocoop/documentation/blob/5427f95417f51d7111a9b8c757fba5669411c350/docs/overview.adoc

To explain why: for simple documentation like this discussion, markdown often suffices. But bylaws, articles of association, and anything remotely legal probably benefits from being able to number things more flexibly than just supporting contiguous numbered lists. As well as cross-document named references, sections, indexes etc., and applying style and alignment hints, without resorting to HTML.

We did at one point need to provide articles and other documents in a court scenario, and paragraph numbering was a particular requirement in that situation. And it is a useful thing to be able to do in general, in fact. And I looked into doing this in Markdown, but the solutions all require stepping beyond it - it's much easier in AsciiDoc (e.g. see [here][1])

[1]: https://github.com/sepheocoop/documentation/blob/5427f95417f51d7111a9b8c757fba5669411c350/Rakefile.rb#L40

AsciiDoc is also rendered in a nice way by GitHub, much like Markdown (although GitHub may ignore some of the styling).  Generating web pages and PDFs from it is also part of its design, the [rakefile][1] does the latter currently, but it could also do the former.

**Reply by wu-lee** [reply]
Just to answer the questions more explicitly

> - What should be made public

I agree with what John said ^^

> - What ultimate presentation for this info to use (wiki or custom)

I'd suggest something like a static-site generator, which turns the raw documentation into a website, as well as into PDF where appropriate. I'd be surprised if Eleventy, which is being used for the website, can't do this. The main snag for that is that it's Javascript, whereas the main AsciiDoc implementations are in Python (originally) and Ruby (latterly, and this is the more maintained version).

Having said that: https://docs.asciidoctor.org/asciidoctor.js/latest/

So: probably custom?

**Reply by King-Mob** [reply]
As I understand it, `.adoc` is a superset of `.md` right?

**Reply by wu-lee** [reply]
Nope, see: https://docs.asciidoctor.org/asciidoc/latest/asciidoc-vs-markdown/

**Comment by shles** [comment]
To combine ideas from @wu-lee  and @King-Mob comments:

I like the idea of putting the docs in the website repo.
Repo is a good place to **edit** and **see the edits history**.
It also allows us to **propose edits privately and keep the uncooked parts hidden**.

Assuming converting .adoc files into pages is fairly straightforward, it satisfies the point of choosing a fairly low-effort solution.

The only question I have now is how to maintain the links between documents. For example, if I want to reference a separate page with a matchmaker role description, what should I put as a URL? Should I use an explicit URL that will be available only after publishing this matchmaker page document, or is there an internal linkage between documents in the repo?

I will now go ahead and create a pull request with documents to the website repo.

I would need help from someone who works on the website to make those docs convertible and publish as pages. 
@wu-lee @King-Mob @spaghettidev123 who can help me?

**Reply by wu-lee** [reply]
> if I want to reference a separate page with a matchmaker role description, what should I put as a URL? Should I use an explicit URL that will be available only after publishing this matchmaker page document, or is there an internal linkage between documents in the repo?

Overall, I think a good rule of thumb would be to prefer relative links where possible, or links relative to the site base.  

I think mixing adoc and markdown (and html) might get complicated.  AsciiDoc can create indexes and so on, but isn't really aware of markdown; and markdown just isn't aware of anything much, nor is HTML.

So possibly we might need to navigate the implications of this. If everything is AsciiDoc then it's relatively simple I think, but if there's a mixture then it's probably possible, but could be fragile with respect to moving or renaming pages.  In which case the AsciiDoc sections might want to be considered opaque blobs of HTML or PDF which get inserted into the site.

So I was imagining that we'd have PDF versions of organisational documents like the article for printing, and HTML versions for reading online. But possibly everything could be AsciiDoc? And then the sub-documents could be included with that relatively seamlessly - perhaps?

But then that seems at odds with using Eleventy... basically I, or someone, would need to look into that.

**Reply by wu-lee** [reply]
Eleventy-wise, there is this: https://github.com/saneef/eleventy-plugin-asciidoc

The docs say this can allow Eleventy use .adoc files instead of .md files.  As I understand it, AsciiDoc already has a mechanism for meta-data like the "front-matter" YAML normally used by Eleventy, stitched onto the markdown content, so perhaps this would allow .adoc pages to work seamlessly with the website as a whole.

Trial and error needed...

**Reply by wu-lee** [reply]
Also: https://kawcco.com/weblog/2023-06-02-hacking-together-a-blog/
